libr/arch/arch_session.c

```diff
@@ -0,0 +1,197 @@
+/* radare - LGPL - Copyright 2020 - pancake */
+
+#include <r_arch.h>
+#include <r_asm.h>
+#include "../config.h"
+
+
+static bool r_arch_session_set_cpu(RArchSession *ai, const char *cpu) {
```

Comment:
I don't think the session should have any of these setters, but be immutable wrt. its setup. Otherwise what is even the point of the lazy session?

Reply:
thats static, its called when creating the session. and lazy is who calls the session consstructor. so yes, its immutable